OCD-4783: Endpoints for new system admin pages

kekey1 opened the pull request.

kekey1 committed:
refactor: Move CQM-related classes into cqm package

kekey1 committed:
refactor: Remove unused CQM cache

[#OCD-4783]

kekey1 committed:
feat!: Create new GET /cqms endpoint

[#OCD-4783]

andlar commented on chpl/chpl-api/src/main/java/gov/healthit/chpl/web/controller/CqmCriteriaController.java:
```diff
@@ -0,0 +1,38 @@
+@Tag(name = "cqms", description = "Endpoints related to Clinical Quality Measures.")
+@RestController
+@RequestMapping("/cqms")
+public class CqmCriteriaController {
```
Should `Criteria` be in this controller name / filename / etc.?

kekey1 replied:
I don't know. For that matter, is it really "eCqms"? We have really inconsistent naming of this throughout. The database table is called cqm_criterion. In many places in the code we call one of these a CqmCriterion. In many other places we do not. There are plenty of variable and method names where it is simply cqm/cqms. When I made the class I was thinking of it as something analogous to the CertificationCriteriaController, so I went with CqmCriteriaController. I also sometimes get the term "CQM Criterion" confused with the 4 c-criteria that can be associated with any attested CQM.

kekey1 committed:
Merge branch 'staging' into OCD-4783

kekey1 committed:
Merge branch 'staging' into OCD-4783

kekey1 committed:
refactor: Remove "criteria" from newly added CQM classes

We think the correct term is simply "CQMs" and not "CQM Criteria", which
unfortunately permeates throughout our code and data model.

[#OCD-4783]

kekey1 committed:
Merge branch 'staging' into OCD-4783

kekey1 committed:
Merge branch 'staging' into OCD-4783

kekey1 committed:
refactor: Remove test data DTOs; move code into testdata package

kekey1 committed:
feat!: Create GET /test-data endpoint

[#OCD-4783]

kekey1 committed:
Merge branch 'staging' into OCD-4783

kekey1 committed:
refactor: Remove caching of deprecated test data

[#OCD-4783]
